components/org.wso2.carbon.identity.recovery/src/main/java/org/wso2/carbon/identity/recovery/internal/service/impl/password/PasswordRecoveryManagerImpl.java

```diff
@@ -910,17 +910,27 @@ private PasswordResetCodeDTO validateConfirmationCode(UserAccountRecoveryManager
 
         UserRecoveryData userRecoveryData;
         try {
-            userRecoveryData = userAccountRecoveryManager.getUserRecoveryData(confirmationCode,
+            String hashedConfirmationCode = Utils.hashCode(confirmationCode);
+            userRecoveryData = userAccountRecoveryManager.getUserRecoveryData(hashedConfirmationCode,
                     RecoverySteps.UPDATE_PASSWORD);
+        } catch (NoSuchAlgorithmException e) {
+            throw Utils.handleServerException(
+                    IdentityRecoveryConstants.ErrorMessages.ERROR_CODE_NO_HASHING_ALGO_FOR_CODE, null);
         } catch (IdentityRecoveryException e) {
-            if (IdentityRecoveryConstants.ErrorMessages.ERROR_CODE_INVALID_RECOVERY_CODE.getCode().equals(
-                    e.getErrorCode())) {
-                e.setErrorCode(IdentityRecoveryConstants.ErrorMessages.ERROR_CODE_INVALID_RECOVERY_FLOW_ID.getCode());
+            try {
+                userRecoveryData = userAccountRecoveryManager.getUserRecoveryData(confirmationCode,
+                        RecoverySteps.UPDATE_PASSWORD);
+            } catch (IdentityRecoveryException ex) {
+                if (IdentityRecoveryConstants.ErrorMessages.ERROR_CODE_INVALID_RECOVERY_CODE.getCode().equals(
+                        ex.getErrorCode())) {
+                    ex.setErrorCode(IdentityRecoveryConstants.ErrorMessages.ERROR_CODE_INVALID_RECOVERY_FLOW_ID.getCode());
+                }
+                throw ex;
             }
-            throw e;
         }
-        if (!StringUtils.equals(userRecoveryData.getRemainingSetIds(),
-                NotificationChannels.EMAIL_CHANNEL.getChannelType())) {
+        if (!(StringUtils.equals(userRecoveryData.getRemainingSetIds(),NotificationChannels.EMAIL_CHANNEL.
+                getChannelType()) || StringUtils.equals(userRecoveryData.getRemainingSetIds(),NotificationChannels.
+                EXTERNAL_CHANNEL.getChannelType()))) {
```

Comment:
```suggestion
        if (!(StringUtils.equals(userRecoveryData.getRemainingSetIds(), NotificationChannels.EMAIL_CHANNEL.
                getChannelType()) || StringUtils.equals(userRecoveryData.getRemainingSetIds(), NotificationChannels.
                EXTERNAL_CHANNEL.getChannelType()))) {
```
formatting issue

Better to use <Constant>.equals(<variable>) format

Reply:
Fixed with 6e6aa3151db682c8b69adc0ed7e8f068a8e713e2